src/include/sof/audio/dp_queue.h

```diff
@@ -159,6 +159,8 @@ struct dp_queue *dp_queue_create(size_t min_available, size_t min_free_space, ui
 static inline
 void dp_queue_free(struct dp_queue *dp_queue)
 {
+	if (!dp_queue)
+		return;
```

Comment:
in principle it's probably good, since this makes the function compatible with `free()`, `rfree()`, etc., but just wondering - can this actually happen? Do we ever have a chance of calling it with `NULL`?

Reply:
Yes, it can, very often when changes from this PR are applied:
```
#if CONFIG_ZEPHYR_DP_SCHEDULER
	dp_queue_free(buffer->stream.dp_queue_sink);
	dp_queue_free(buffer->stream.dp_queue_source);
#endif /* CONFIG_ZEPHYR_DP_SCHEDULER */
```

there's no condition if freed pointers are not null, and those will be nulls every time the buffer is no shadowed (so - in most cases)